backend/api/data.py

```
@@ -118,7 +119,8 @@ class PostAnalyticsResponse(BaseModel):
 @router.post("/posts", response_model=DataQueryResponse)
 async def query_posts(
     query: PostQueryRequest,
-    db: Session = Depends(get_db)
+    db: Session = Depends(get_db),
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)